ha/core/node/replacement/refresh_context.py

```diff
@@ -76,16 +76,28 @@ def is_cleanup_required(self, node=None):
                 status_list[resource] = self._decision_monitor.get_resource_status(resource)
             else:
                 pass
-        Log.info(f"Resource status for node {node} is {status_list}")
         if Action.FAILED in status_list.values():
-            Log.debug("Some component are not yet recovered skipping failback")
+            Log.debug(f"Some component are not yet recovered skipping failback. status: {status_list}")
         elif Action.RESOLVED in status_list.values():
-            Log.info("Failback is required as some of alert are resolved.")
+            Log.info(f"Failback is required as some of alert are resolved. status: {status_list}")
+            self._ack_resource(status_list)
```

Comment:
Couple of scenarios -
1. There are multiple types of fault and one of these is resolved. Then even after acknowledgement, the failback should not be initiated.
2. Can decision monitor raises any exception while acknowledge_resource is called ? If yes, the we need to handle the scenario.

Please check if above two cases are handled her.

Reply:
```1. There are multiple types of fault and one of these is resolved. Then even after acknowledgement, the failback should not be initiated.```
- Here failback and ack only happen if all fault are resolved, even 4 out 3 resolved failback will not happen.
It take all alert status and if all are resolved then only trigger failback.

```2. Can decision monitor raises any exception while acknowledge_resource is called ? If yes, the we need to handle the scenario.```
If decision maker raise new alert occurs during ack then ack check latest alert and resolve only if latest is resolved
But as refresh class already stored resolved then it perform cleanup and which lead to failback and failover.
But not stuck in cycle.

Reply:
As per our discussion, these scenarios are handled.